frontend/src/views/WheresMyGene/components/GeneSearchBar/components/SaveExport/index.tsx

```diff
@@ -190,8 +211,12 @@ export default function SaveExport({
 
     observer.observe(heatmapNode, config);
 
+    // This will make a change to the heatmap dom which triggers the observer to start the download
     if (selectedFileTypes.includes("svg")) {
       setEchartsRendererMode("svg");
+    } else {
+      // Kind of a hack to modify the DOM to trigger the observer
+      heatmapNode.classList.add("is-downloading");
```

Comment:
The observer listens to changes in the dom, and once detected it will be triggered and then the download still start after `MUTATION_OBSERVER_TIMEOUT` seconds. We only do this for PNG and CSV because `echartsRendererMode` only needs to be set for SVG, and changing the renderer mode also triggers the download.

Reply:
I think @seve was saying PNG quality is better when echarts is in SVG mode vs. PNG mode? If that's true, then I think only CSV download needs this hack!